packages/components/src/components/card/card.component.ts

```diff
@@ -0,0 +1,205 @@
+    *
+    */
+   @queryAssignedElements({ slot: 'footer-button-secondary' })
+   footerButtonSecondary?: Array<HTMLElement>;
+
+   override update(changedProperties: PropertyValues<Card>) {
+     super.update(changedProperties);
+     if (changedProperties.has('variant')) {
+       this.updateFooterButtonColors();
+     }
+   }
+
+   /**
+    * Handles the icon buttons in the header section and sets its variant for styling.
+    * It also limits the number of buttons to 3.
+    * @internal
+    */
+   private handleIconButtons = () => {
+     if (this.iconButtons) {
+       this.iconButtons.forEach((element) => {
+         if (!element.matches(DEFAULTS.BUTTON) && element.getAttribute('data-btn-type') !== 'icon') {
+           element.remove();
+         } else {
+           element.setAttribute('variant', 'tertiary');
+           element.setAttribute('size', '32');
+         }
+       });
+     }
+     // limit to show only first 3 buttons defined in the slot
+     if (this.iconButtons && this.iconButtons.length > 3) {
+       for (let i = 3; i < this.iconButtons.length; i += 1) {
+         this.iconButtons[i].remove();
+       }
+     }
+   };
+
+   /**
+    * Filters and renders only the following content into the footer section and removes anything other than it
+    * - One mdc-link element in the footer-link slot
+    * - One secondary variant of the mdc-button element in the footer-button-secondary slot
+    * - One primary variant of the mdc-button element in the footer-button-primary slot
+    *
+    * @internal
+    */
+ private handleFooterSlot = (tagname: string, variant = '') => {
+   let arrayItems: Array<HTMLElement> = [];
+   if (tagname === DEFAULTS.LINK && this.footerLink?.length) {
+     arrayItems = this.footerLink;
+   } else if (tagname === DEFAULTS.BUTTON && variant === 'primary' && this.footerButtonPrimary?.length) {
+     arrayItems = this.footerButtonPrimary;
+   } else if (tagname === DEFAULTS.BUTTON && variant === 'secondary' && this.footerButtonSecondary?.length) {
+     arrayItems = this.footerButtonSecondary;
+   }
+   // if there are more than one instance, remove them.
+   if (arrayItems.length > 1) {
+     for (let i = 1; i < arrayItems.length; i += 1) {
+       arrayItems[i].remove();
+     }
+   }
+   arrayItems.forEach((element) => {
+     // remove the element if it doesn't match with the tagname
+     if (!element.matches(tagname)) {
+       element.remove();
+     }
+     // set the variant if it is provided
+     if (variant) {
+       element.setAttribute('variant', variant);
+     }
+   });
+ };
+
+ /**
+  * Updates the color of the footer buttons based on the variant.
+  * If the variant is promotional, the color is promotional, else default.
+  *
+  * @internal
+  */
+ private updateFooterButtonColors = () => {
+   const footerButtons = [...(this.footerButtonPrimary || []), ...(this.footerButtonSecondary || [])];
+   if (footerButtons.length > 0) {
+     footerButtons.forEach((button) => {
```

Comment:
The if condition is not needed, as forEach will not work if the array is empty.

```suggestion
     footerButtons.forEach((button) => {
```